types.go

```diff
@@ -89,11 +89,14 @@ type cfgString struct {
 	s string
 }
 
+type cfgSubFlex struct{ cfgSub }
```

Comment:
When should we use cfgSubFlex? Can this hide developers passing the wrong types, because errors are not returned if there was a type mismatch (e.g. users tries to unpack an object into an int)?

Reply:
I would be interested to talk to you about why I had to add this and if there was another way of doing it. But it is specifically needed to fix 1 test `TestReifyUnpackersPtr`. This test uses pointers to the structures which causes a specific path in reify to be executed that causes the configuration to be wrapped in `cfgSub`.

Then unpack tries to use `cfgSub` to fill in a type that `cfgSub` returns a `ErrTypeMismatch`.